api/api_test.go

```diff
@@ -98,29 +99,85 @@ func TestServeAssets(t *testing.T) {
 	}
 }
 
-func TestNotify(t *testing.T) {
-	api := New(config, context.TODO())
+func TestWebsocketNotify(t *testing.T) {
+	api := New(config)
+	server := httptest.NewServer(api)
+
+	firstConnection, err := createWebsocket(server)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	secondConnection, err := createWebsocket(server)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// First message received is the connected message which can be ignored
+	firstConnection.ReadJSON(&StatusUpdate{})
+	secondConnection.ReadJSON(&StatusUpdate{})
+
 	expectedUpdate := StatusUpdate{Type: "Some message", Extensions: config.Extensions}
+	api.Notify(expectedUpdate)
 
-	go api.Notify(expectedUpdate)
+	if err := verifyWebsocketMessage(firstConnection, expectedUpdate); err != nil {
+		t.Error(err)
+	}
 
-	update := <-api.notifier.updates
+	if err = verifyWebsocketMessage(secondConnection, expectedUpdate); err != nil {
+		t.Error(err)
+	}
+}
 
-	if update.Type != expectedUpdate.Type {
-		t.Errorf("Unexpected Type in update event, received %v, expected %v", update.Type, expectedUpdate.Type)
+func TestWebsocketConnection(t *testing.T) {
+	api := New(config)
+	server := httptest.NewServer(api)
+	ws, err := createWebsocket(server)
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	expectedResult, err := json.Marshal(expectedUpdate.Extensions)
+	if err := verifyWebsocketMessage(ws, StatusUpdate{Type: "connected", Extensions: api.Extensions}); err != nil {
+		t.Error(err)
+	}
+
+	api.Shutdown()
```

Comment:
I've reworked our `Shutdown` mechanism. While it works now, we're not doing 100% the right thing just yet. The connection is currently considered to have terminated abnormally. In a follow up PR, we should make sure to send a proper closing message that leads to a controlled shutdown. Among other things, an abnormal shutdown prevents `CloseHandler`s from being executed.